xbmc/storage/MediaManager.cpp

```diff
@@ -64,23 +64,21 @@ const char MEDIA_SOURCES_XML[] = { "special://profile/mediasources.xml" };
 CMediaManager::CMediaManager()
 {
   m_bhasoptical = false;
-  m_platformStorage = nullptr;
 }
 
 void CMediaManager::Stop()
 {
   if (m_platformStorage)
     m_platformStorage->Stop();
 
-  delete m_platformStorage;
-  m_platformStorage = nullptr;
+  m_platformStorage.reset();
 }
 
 void CMediaManager::Initialize()
 {
   if (!m_platformStorage)
   {
-    m_platformStorage = IStorageProvider::CreateInstance();
+    m_platformStorage.reset(IStorageProvider::CreateInstance());
```

Comment:
Can you go further and make `IStorageProvider::CreateInstance()` return a unique_ptr?

This will require updating the platform methods also.

Reply:
Maybe later, my goal here was to understand how the platform based "components" are initialized. Will keep it in mind (and ping you if I am not able to sort it out) when doing the same for disc handling.

Reply:
It really wouldnt take much more effort. It would also help keep the use of smart pointers consistent rather than initializing a smart pointer from a raw pointer.